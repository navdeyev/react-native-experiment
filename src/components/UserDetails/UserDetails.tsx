import * as React from 'react';
import {Text, View} from 'react-native';

import {User} from 'src/domains/types';

interface Props {
    user: User;
}

const UserDetails: React.SFC<Props> = props => {
    const {user} = props;
    return (
        <View>
            <Text>Name: {user.name}</Text>
            <Text>Username: {user.username}</Text>
            <Text>Email: {user.email}</Text>
            <Text>Phone: {user.phone}</Text>
        </View>
    );
};

export default UserDetails;
