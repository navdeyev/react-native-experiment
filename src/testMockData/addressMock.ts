import {Address, Geo} from 'src/domains/types';

export const geo: Geo = {
    lat: '-37.3159',
    lng: '81.1496'
};

export const address: Address = {
    street: 'Kulas Light',
    suite: 'Apt. 556',
    city: 'Gwenborough',
    zipcode: '92998-3874',
    geo
};
