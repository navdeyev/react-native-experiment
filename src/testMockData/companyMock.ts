import {Company} from 'src/domains/types';

export const company: Company = {
    name: 'Romaguera-Crona',
    catchPhrase: 'Multi-layered client-server neural-net',
    bs: 'harness real-time e-markets'
};
