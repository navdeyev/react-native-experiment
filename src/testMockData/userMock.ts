import {User} from '../domains/types';

import {address} from './addressMock';
import {company} from './companyMock';

export const user: User = {
    id: 1,
    name: 'Leanne Graham',
    username: 'Bret',
    email: 'Sincere@april.biz',
    address,
    phone: '1-770-736-8031 x56442',
    website: 'hildegard.org',
    company
};
