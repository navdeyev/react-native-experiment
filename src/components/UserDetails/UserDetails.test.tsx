import * as React from 'react';
import {shallow} from 'enzyme';

import {user} from 'src/testMockData/userMock';

import UserDetails, {Props} from './UserDetails';

describe('UserDetails', () => {

    let props: Props;
    beforeEach(() => {
        props = {
            user,
            onEditUser: jest.fn()
        }
    });

    it('renders a component', () => {
        const render = shallow(<UserDetails {...props}/>);
        expect(render.getElement()).toMatchSnapshot();
    });

    it('executes onEditUser when edit button was pressed', () => {
        const render = shallow(<UserDetails {...props}/>);
        const btn = render.find('[testID="edit-user-details-button"]');
        btn.simulate('press');
        expect(props.onEditUser).toHaveBeenCalledWith(props.user);
    });
});