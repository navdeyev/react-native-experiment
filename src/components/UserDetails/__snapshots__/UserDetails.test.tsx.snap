// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails renders a component 1`] = `
<Component>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Name: 
    Leanne Graham
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Username: 
    Bret
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Email: 
    Sincere@april.biz
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Phone: 
    1-770-736-8031 x56442
  </Text>
</Component>
`;
